packages/gatsby-theme-api-docs/src/components/type/type.js

```diff
@@ -12,13 +12,44 @@ import Enum from './enum';
 import Properties from './properties/properties';
 import Examples from './examples';
 
+const contentTypeToPrimitiveMap = {
+  'application/json': {
+    number: '_Any JSON "number"',
+    any: '_Any valid JSON_',
+    object: '_Any JSON "object"_',
+    boolean: '_Any JSON "boolean"_',
+    string: '_Any JSON "string"_',
+    array: '_Any JSON "array"_',
```

Comment:
Since the only formatting we chose to use here is making the whole thing italic and the values are hardwired in the codebase I suggest to leave out the (relatively expensive) markdownFragmentToReact  step and format the output natively using React.  The components representing the various markdown features are all available in the docs-ui-kit.    
If you find it more expressive, the map here could alternatively return a component so you can use JSX for the values directly there.  But I would rather not overdo it if there is no need for more formatting.